src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = String(this[0]);
+
+    if (this[0] === null || this[0] === undefined) {
+      result = '';
+    }
```

Comment:
Instead of this code, you can just create empty result string and iterate from first element instead of second in your loop. In case if array is empty - your loop wouldn't start and function will return empty string